code/modules/paperwork/paperbin.dm

```diff
@@ -23,14 +23,23 @@
 /obj/item/weapon/paper_bin/getFireFuel()
 	return amount
 
-/obj/item/weapon/paper_bin/MouseDrop(mob/user as mob)
-	if((user == usr && (!( usr.restrained() ) && (!( usr.stat ) && (usr.contents.Find(src) || in_range(src, usr))))))
+/obj/item/weapon/paper_bin/MouseDrop(over_object)
+	if(!usr.restrained() && !usr.stat && (usr.contents.Find(src) || Adjacent(usr)))
 		if(!istype(usr, /mob/living/carbon/slime) && !istype(usr, /mob/living/simple_animal))
-			if( !usr.get_active_hand() )		//if active hand is empty
-				src.loc = user
-				user.put_in_hands(src)
-				user.visible_message("<span class='notice'>[user] picks up the [src].</span>", "<span class='notice'>You grab [src] from the floor!</span>")
-
+			if(istype(over_object,/obj/screen)) //We're being dragged into the user's UI...
```

Comment:
if you're going to check the name don't bother with a type check and just do a null sanity check